sdk-wallet/keystores/encrypted_keystore.md

```diff
@@ -1,33 +1,48 @@
 ## EncryptedKeystore
 
+`EncryptedKeystore` handles encrypted JSON files.
+
+Note: current design is suboptimal. `EncryptedKeystore` handles both legacy keystore files (that hold encrypted secret keys) and new keystore files (that hold encrypted mnemonics). `get_kind()` can be used as a differentiator. A future design might involve two separate classes.
+
 ```
 class EncryptedKeystore:
-    // The constructor is not captured by the specs; it's up to the implementing library to define it.
+    // The constructor is not strictly captured by the specs; it's up to the implementing library to define it. Suggestion below.
```

Comment:
Please remove this private constructor, it's out of the scope of the provided specs. We should have only those 2 named constructors as example
+ providing a type on which the constructor will do a switch is not a very good example (clean code wise)

Reply:
Fixed phrasing, removed private constructor. No switch should be made in the implementation.